Source/WebCore/page/scrolling/mac/ScrollerPairMac.h

```diff
@@ -86,29 +93,36 @@ class ScrollerPairMac {
     // Only for use by WebScrollerImpPairDelegateMac. Do not use elsewhere!
     ScrollerMac& verticalScroller() { return m_verticalScroller; }
     ScrollerMac& horizontalScroller() { return m_horizontalScroller; }
-    
+
     String scrollbarStateForOrientation(ScrollbarOrientation) const;
 
+    ScrollbarStyle scrollbarStyle() const;
+    void setScrollbarStyle(ScrollbarStyle);
+
+    void setVerticalScrollerImp(NSScrollerImp *);
+    void setHorizontalScrollerImp(NSScrollerImp *);
+
 private:
+    ScrollerPairMac(ScrollingTreeScrollingNode&);
+
     NSScrollerImp *scrollerImpHorizontal() { return horizontalScroller().scrollerImp(); }
     NSScrollerImp *scrollerImpVertical() { return verticalScroller().scrollerImp(); }
-    
-    NSScrollerImpPair *scrollerImpPair() { return m_scrollerImpPair.get(); }
-    Lock& scrollerImpPairLock() WTF_RETURNS_LOCK(m_scrollerImpPairLock) { return m_scrollerImpPairLock; }
 
-    bool m_inLiveResize;
-    WebCore::ScrollingTreeScrollingNode& m_scrollingNode;
+    void ensureOnMainThreadWithImpPair(Function<void(NSScrollerImpPair *)>&&);
+
+    bool m_inLiveResize { false };
+    ScrollingTreeScrollingNode& m_scrollingNode;
 
     ScrollerMac m_verticalScroller;
     ScrollerMac m_horizontalScroller;
 
-    WebCore::IntPoint m_lastKnownMousePosition;
-    std::optional<WebCore::FloatPoint> m_lastScrollOffset;
+    IntPoint m_lastKnownMousePosition;
+    std::optional<FloatPoint> m_lastScrollOffset;
 
-    mutable Lock m_scrollerImpPairLock;
+    mutable RecursiveLock m_scrollerImpPairLock;
```

Comment:
Do we still need the lock at all?

Reply:
Good point! I think we can just remove it, since:

- Initialization (`init()`) happens on the main thread.
- We can just cache the last scrollbar style and return that, in `ScrollerPairMac::scrollbarStyle()`.

That should simplify the code a lot, too.